x/concentrated-liquidity/lp_test.go

```diff
@@ -2,66 +2,252 @@ package concentrated_liquidity_test
 
 import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	types "github.com/osmosis-labs/osmosis/v12/x/concentrated-liquidity/types"
 )
 
-func (s *KeeperTestSuite) TestCreatePosition() {
-	denom0 := "eth"
-	denom1 := "usdc"
+type lpTest struct {
+	poolId          uint64
+	owner           sdk.AccAddress
+	currentTick     sdk.Int
+	lowerTick       int64
+	upperTick       int64
+	currentSqrtP    sdk.Dec
+	amount0Desired  sdk.Int
+	amount0Expected sdk.Int
+	amount1Desired  sdk.Int
+	amount1Expected sdk.Int
+	liquidityAmount sdk.Dec
+	expectedError   error
+}
 
-	tests := map[string]struct {
-		poolId            uint64
-		currentTick       sdk.Int
-		lowerTick         int64
-		upperTick         int64
-		currentSqrtP      sdk.Dec
-		amount0Desired    sdk.Int
-		amount0Expected   sdk.Int
-		amount1Desired    sdk.Int
-		amount1Expected   sdk.Int
-		expectedLiquidity sdk.Dec
-	}{
-		"happy path": {
-			poolId:            1,
-			currentTick:       sdk.NewInt(85176),
-			lowerTick:         int64(84222),
-			upperTick:         int64(86129),
-			currentSqrtP:      sdk.MustNewDecFromStr("70.710678118654752440"), // 5000
-			amount0Desired:    sdk.NewInt(1000000),                            // 1 eth
-			amount0Expected:   sdk.NewInt(998587),                             // 0.998587 eth
-			amount1Desired:    sdk.NewInt(5000000000),                         // 5000 usdc
-			amount1Expected:   sdk.NewInt(5000000000),                         // 5000 usdc
-			expectedLiquidity: sdk.MustNewDecFromStr("1517818840.967515822610790519"),
+var (
+	denom0   = "eth"
+	denom1   = "usdc"
+	baseCase = &lpTest{
+		poolId:          1,
+		currentTick:     sdk.NewInt(85176),
+		lowerTick:       int64(84222),
+		upperTick:       int64(86129),
+		currentSqrtP:    sdk.MustNewDecFromStr("70.710678118654752440"), // 5000
+		amount0Desired:  sdk.NewInt(1000000),                            // 1 eth
+		amount0Expected: sdk.NewInt(998587),                             // 0.998587 eth
+		amount1Desired:  sdk.NewInt(5000000000),                         // 5000 usdc
+		amount1Expected: sdk.NewInt(5000000000),                         // 5000 usdc
+		liquidityAmount: sdk.MustNewDecFromStr("1517818840.967515822610790519"),
```

Comment:
nit: maybe "liquidityAmountExpected" or "expectedLiquidityAmount"

Reply:
The thing is it's not always used as "expected".

For `withdrawPosition` test this is an input:
https://github.com/osmosis-labs/osmosis/blob/7d8c224f583d5eb57d08a77ada3bb98e0e4cc671/x/concentrated-liquidity/lp_test.go#L204

That's why I chose a more general name